Sources/NIOCore/EventLoop.swift

```diff
@@ -523,6 +523,50 @@ extension TimeAmount: AdditiveArithmetic {
     }
 }
 
+#if swift(>=5.7)
+extension TimeAmount {
+    @available(macOS 13, iOS 16, tvOS 16, watchOS 9, *)
+    /// Creates a new `TimeAmount` for the given `Duration`, truncating and clamping if necessary.
+    ///
+    /// - returns: `TimeAmount`, truncated to nanosecond precision, and clamped to `Int64.max` nanoseconds.
+    public init(_ duration: Swift.Duration) {
+        self = .nanoseconds(duration.nanosecondsClamped)
+    }
+}
+
+@available(macOS 13, iOS 16, tvOS 16, watchOS 9, *)
+extension Swift.Duration {
+    /// Construct a `Duration` given a number of nanoseconds represented as a `TimeAmount`.
+    ///
+    /// - returns: A `Duration` representing a given number of nanoseconds.
+    public init(_ timeAmount: TimeAmount) {
+        self = .nanoseconds(timeAmount.nanoseconds)
+    }
+}
+
+@available(macOS 13, iOS 16, tvOS 16, watchOS 9, *)
+internal extension Swift.Duration {
+    /// The duration represented as nanoseconds, clamped to maximum expressible value.
+    var nanosecondsClamped: Int64 {
+        let components = self.components
+
+        let secondsComponentNanos = components.seconds.multipliedReportingOverflow(by: 1_000_000_000)
+        let attosCompononentNanos = components.attoseconds.dividedReportingOverflow(by: 1_000_000_000)
```

Comment:
I don't think we need `dividedReportingOverflow` here: we have a statically known quotient so we know this cannot overflow.